src/test/consensus/Validations_test.cpp

```diff
@@ -777,52 +817,212 @@ class Validations_test : public beast::unit_test::suite
+        BEAST_EXPECT(
+            harness.vals().getPreferredLCL(ledgerB, Seq{2}, peerCounts) ==
+            ledgerB.id());
+    }
+
+    void
+    testAcquireValidatedLedger()
+    {
+        using namespace std::chrono_literals;
+        testcase("Acquire validated ledger");
+
+        LedgerHistoryHelper h;
+        TestHarness harness(h.oracle);
+        Node a = harness.makeNode();
+
+        using ID = Ledger::ID;
+        using Seq = Ledger::Seq;
+
+        // Validate the ledger before it is actually available
+        Validation val = a.validate(ID{2}, Seq{2}, 0s, 0s, true);
+
+        BEAST_EXPECT(ValStatus::current == harness.add(val));
+        // Validation is available
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ID{2}) == 1);
+        // but ledger based data is not
+        BEAST_EXPECT(harness.vals().getNodesAfter(genesisLedger, ID{0}) == 0);
+
+        // Create the ledger
+        Ledger ledgerAB = h["ab"];
+        // Now it should be available
+        BEAST_EXPECT(harness.vals().getNodesAfter(genesisLedger, ID{0}) == 1);
+
+        // Create a validation that is not available
+        harness.clock().advance(5s);
+        Validation val2 = a.validate(ID{4}, Seq{4}, 0s, 0s, true);
+        BEAST_EXPECT(ValStatus::current == harness.add(val2));
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ID{4}) == 1);
+        BEAST_EXPECT(
+            harness.vals().getPreferred(genesisLedger) ==
+            std::make_pair(ledgerAB.seq(), ledgerAB.id()));
+
+        // Switch to validation that is available
+        harness.clock().advance(5s);
+        Ledger ledgerABCDE = h["abcde"];
+        BEAST_EXPECT(ValStatus::current == harness.add(a.partial(ledgerABCDE)));
+        BEAST_EXPECT(
+            harness.vals().getPreferred(genesisLedger) ==
+            std::make_pair(ledgerABCDE.seq(), ledgerABCDE.id()));
+    }
+
+    void
+    testNumTrustedForLedger()
+    {
+        testcase("NumTrustedForLedger");
+        LedgerHistoryHelper h;
+        TestHarness harness(h.oracle);
+        Node a = harness.makeNode();
+        Node b = harness.makeNode();
+        Ledger ledgerA = h["a"];
+
+        BEAST_EXPECT(ValStatus::current == harness.add(a.partial(ledgerA)));
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerA.id()) == 0);
+
+        BEAST_EXPECT(ValStatus::current == harness.add(b.validate(ledgerA)));
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerA.id()) == 1);
+    }
+
+    void
+    testSeqEnforcer()
+    {
+        testcase("SeqEnforcer");
+        using Seq = Ledger::Seq;
+        using namespace std::chrono;
+
+        beast::manual_clock<steady_clock> clock;
+        SeqEnforcer<Seq> enforcer;
+
+        ValidationParms p;
+
+        BEAST_EXPECT(enforcer(clock.now(), Seq{1}, p));
+        BEAST_EXPECT(enforcer(clock.now(), Seq{10}, p));
+        BEAST_EXPECT(!enforcer(clock.now(), Seq{9}, p));
+        BEAST_EXPECT(!enforcer(clock.now(), Seq{5}, p));
```

Comment:
You could swap the `9` and `5` checks to confirm that 9 fails after 5 even though it's >5 (but they're both <10)